Userland/Libraries/LibJS/AST.cpp

```diff
@@ -848,38 +794,40 @@ Value ForInStatement::execute(Interpreter& interpreter, GlobalObject& global_obj
     });
 
     auto last_value = js_undefined();
+
     while (object) {
-        auto property_names = TRY_OR_DISCARD(object->enumerable_own_property_names(Object::PropertyKind::Key));
+        auto property_names = TRY(object->enumerable_own_property_names(Object::PropertyKind::Key));
         for (auto& value : property_names) {
-            TRY_OR_DISCARD(for_in_head_state.execute_head(interpreter, global_object, value));
-            last_value = m_body->execute(interpreter, global_object).value_or(last_value);
+            TRY(for_in_head_state.execute_head(interpreter, global_object, value));
+            auto result = m_body->execute(interpreter, global_object);
             interpreter.vm().running_execution_context().lexical_environment = old_environment;
-            if (interpreter.exception())
-                return {};
+            if (auto* exception = interpreter.exception())
+                return throw_completion(exception->value());
```

Comment:
Why do we still check exception here? Can we not just check the result? 
~Or even do it with TRY?~ I guess not since we want to know if it is a continue or break.

Reply:
Yep, easier to temporarily do it like this instead of now adding a TRY and then removing it again when we do the fully the right (tm) way.